### PR TITLE
[rfr] From Port and To Port should accept values of 0 

### PR DESCRIPTION
**Pull request by deniszh**
…when the IP Protocol is "ICMP".
Fixing https://github.com/rackspace/gophercloud/issues/521

### CODE REVIEW

**Comment by deniszh** [comment]
Not sure do I need to test that. Tried to hack existing tests but failed.

File: openstack/compute/v2/extensions/secgroups/requests.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package secgroups
 
 import (
 	"errors"
+    "strings"
```
Comment (jtopjian) [inline openstack/compute/v2/extensions/secgroups/requests.go]: spacing issue.

**Comment by deniszh** [comment]
fixed, thanks!

**Comment by deniszh** [comment]
Yay, we have a test now! Thanks, @jtopjian!

**Comment by jtopjian** [comment]
np :smile: 

Just a heads up to the Gophercloud team: there's something preventing security group tests from succeeding. The error seen (without this patch) is:

```
convenience.go:34: Failure in secgroup_test.go, line 160: unexpected error "EOF"
```

Discarding the error check makes everything work.

Also, this patch is required because it looks as though the Contrail Neutron plugin is mangling ICMP rules in OpenStack. Even though `-1/-1` is used as input, `0/65535` is returned.

**Comment by jtopjian** [comment]
@deniszh Can you append an [rfr] tag to the beginning of this PR? That notifies the gophercloud team to review the PR.

@jrperritt if you have time, can you give this a quick review and let us know if it's something that can be merged quickly or needs more thought? All Contrail-based Neutron environments are affected by this.

**Comment by deniszh** [comment]
@jtopjian - like that? I can't put any label to it

**Comment by jtopjian** [comment]
Yup, just like that. :)

**Comment by jrperritt** [comment]
This looks good but needs a unit test.

**Comment by deniszh** [comment]
@jrperritt: it already has test from @jtopjian - but if that is not enough - what test is needed then?

**Comment by jrperritt** [comment]
It has an acceptance test; it needs a unit test. See here:
https://github.com/deniszh/gophercloud/blob/master/openstack/compute/v2/extensions/defsecrules/requests_test.go
and
https://github.com/deniszh/gophercloud/blob/master/openstack/compute/v2/extensions/secgroups/requests_test.go

**Comment by jtopjian** [comment]
@deniszh I've just opened a PR to your local repo with the applicable unit tests.

**Comment by deniszh** [comment]
Wow, @jtopjian, thanks a lot, again!
Was completely busy with other stuff...

Build status is green now.
@jrperritt - does this looks good?

**Comment by jrperritt** [comment]
Thank you @jtopjian and @deniszh . +2

**Pull request merged**